application/cms/classification_service.py

```diff
@@ -0,0 +1,235 @@
+        values = Ethnicity.query.all()
+        for value in values:
+            if len(value.classifications) == 0:
+                db.session.delete(value)
+                db.session.commit()
+
+    """
+    CATEGORY >-< VALUE relationship management
+    """
+
+    def add_value_to_classification(self, classification, value_title):
+        value = self.create_or_get_value(value_string=value_title)
+        classification.values.append(value)
+
+        db.session.add(classification)
+        db.session.commit()
+        return classification
+
+    def add_value_to_classification_as_parent(self, classification, value_string):
+        value = self.create_or_get_value(value_string=value_string)
+        classification.parent_values.append(value)
+
+        db.session.add(classification)
+        db.session.commit()
+        return classification
+
+    def add_values_to_classification(self, classification, value_strings):
+        for value_string in value_strings:
+            value = self.create_or_get_value(value_string=value_string)
+            classification.values.append(value)
+        db.session.add(classification)
+        db.session.commit()
+        return classification
+
+    def add_values_to_classification_as_parents(self, classification, value_strings):
+        for value_string in value_strings:
+            value = self.create_or_get_value(value_string=value_string)
+            classification.parent_values.append(value)
+        db.session.add(classification)
+        db.session.commit()
+        return classification
+
+    def remove_value_from_classification(self, classification, value_string):
+        value = self.get_value(value=value_string)
+
+        classification.values.remove(value)
+        db.session.add(classification)
+        db.session.commit()
+
+    def remove_parent_value_from_classification(self, classification, value_string):
+        value = self.get_value(value=value_string)
+
+        classification.parent_values.remove(value)
+        db.session.add(classification)
+        db.session.commit()
+
+    @staticmethod
+    def remove_classification_values(classification):
+        for value in classification.values:
+            db.session.delete(value)
+        db.session.commit()
+
+    @staticmethod
+    def remove_parent_classification_values(classification):
+        for value in classification.parent_values:
+            db.session.delete(value)
+        db.session.commit()
+
+
+class ClassificationLink:
+    def __init__(self, classification_id, includes_parents=False, includes_all=False, includes_unknown=False):
+        self.classification_id = classification_id
+        self.includes_parents = includes_parents
+        self.includes_all = includes_all
+        self.includes_unknown = includes_unknown
+
+    def get_classification(self):
+        return classification_service.get_classification_by_id(self.classification_id)
+
+    def is_more_complex_than(self, link):
```

Comment:
@TheDoubleK I think this method, and the `__get_classification_complexity` one below, can both now be dropped?